kernel/kernel-api/src/main/java/io/delta/kernel/internal/icebergcompat/IcebergCompatMetadataValidatorAndUpdater.java

```diff
@@ -151,10 +166,127 @@ Optional<Metadata> validateAndUpdate(
    * Defines checks for compatibility with the targeted iceberg features (icebergCompatV1 or
    * icebergCompatV2 etc.)
    */
-  interface IcebergCompatCheck {
+  protected interface IcebergCompatCheck {
     void check(IcebergCompatInputContext inputContext);
   }
 
+  ///////////////////////////////////////////////////////////
+  /// Implementation of {@link IcebergCompatCheck}        ///
+  ///////////////////////////////////////////////////////////
+  protected static final IcebergCompatCheck CHECK_NO_COMPAT_V1_ENABLED =
+      (inputContext) -> {
+        if (Boolean.valueOf(
+            inputContext
+                .newMetadata
+                .getConfiguration()
+                .getOrDefault("delta.enableIcebergCompatV1", "false"))) {
+          throw DeltaErrors.icebergCompatIncompatibleVersionEnabled(
+              inputContext.compatFeatureName, "delta.enableIcebergCompatV1");
+        }
+      };
```

Comment:
how about making this generic where it takes a list of compatV1 or compatV2 and ensures none of them is enabled? From IcebergCompatV2 we pass `icebergCompatV1` and `icebergCompatV3`. From `icebergCompatV3` we pass we pass `icebergCompatV1` and `icebergCompatV2`